nixos/modules/services/databases/lldap.nix

```
@@ -102,12 +102,81 @@ in
           };
+
+          ldap_user_pass_file = mkOption {
+            type = types.nullOr types.str;
+            default = null;
+            description = ''
+              Path to a file containing the default admin password.
+            '';
+          };
+
+          force_ldap_user_pass_reset = mkOption {
+            type = types.oneOf [
+              types.bool
+              (types.enum [ "always" ])
+            ];
+            default = false;
+            description = ''
+              Force reset of the admin password.
+
+              Break glass in case of emergency: if you lost the admin password, you
+              can set this to true to force a reset of the admin password to the value
+              of `ldap_user_pass_file`.
+
+              Alternatively, you can set it to `"always"` to reset every time the server starts
+              which makes for a more declarative configuration.
+
+              The difference between `true` and `"always"` is the former is intended for a one time fix
+              while the latter is intended for a declarative workflow. In practice, the result
+              is the same: the password gets reset. The only practical difference is the former
+              outputs a warning message while the latter outputs an info message.
+            '';
+          };
+
+          jwt_secret_file = mkOption {
+            type = types.nullOr types.str;
+            default = null;
+            description = ''
+              Path to a file containing the JWT secret.
+            '';
+          };
         };
       };
+
+      # TOML does not allow null values, so we use null to omit those fields
+      apply = lib.filterAttrsRecursive (_: v: v != null);
+    };
+
+    silenceForceUserPassResetWarning = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Disable warning when the admin password is set declaratively with `ldap_user_pass_file` option
+        but the `force_ldap_user_pass_reset` is set to `false`.
+
+        This can lead to the admin password to drift from the one given declaratively.
+        If that is okay for you and you want to silence the warning, set this option to `true`.
+      '';
     };
   };
 
   config = lib.mkIf cfg.enable {
+    warnings =
+      lib.optionals
+        (
+          cfg.settings.ldap_user_pass_file != null
+          && cfg.settings.force_ldap_user_pass_reset == false
+          && cfg.silenceForceUserPassResetWarning == false
+        )
+        [
+          ''
+            The default admin password is set declaratively with `ldap_user_pass_file` option but the `force_ldap_user_pass_reset` is set to `false`.
+            This means the admin password can be changed through the UI and will drift from the one defined in your nix config.
+            Either set the `force_ldap_user_pass_reset` option to `true` or `"always"` or silence this warning by setting `silenceForceUserPassResetWarning` to `true`.
```

Right now, the error message makes it hard to know that it came from lldap:

```console
evaluation warning: The default admin password is set declaratively with `ldap_user_pass_file` option but the `force_ldap_user_pass_reset` is set to `false`.
                    This means the admin password can be changed through the UI and will drift from the one defined in your nix config.
                    Either set the `force_ldap_user_pass_reset` option to `true` or `"always"` or silence this warning by setting `silenceForceUserPassResetWarning` to `true`.
```

Some rewordings to mention lldap and maybe to distinguish between settings (lldap) and options (nixos)?

```suggestion
            lldap: The default admin password is declared with the setting `ldap_user_pass_file`, but `force_ldap_user_pass_reset` is set to `false`.
            This means the admin password can be changed through the UI and will drift from the one defined in your nix config.
            Either set `force_ldap_user_pass_reset` to `"always"` (or `true`), or silence this warning by setting the option `services.lldap.silenceForceUserPassResetWarning` to `true`.
```